test/topology_random_failures/test_random_failures.py

```diff
@@ -0,0 +1,212 @@
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+from __future__ import annotations
+
+import sys
+import time
+import random
+import logging
+import itertools
+from typing import TYPE_CHECKING
+from contextlib import suppress
+
+import psutil
+import pytest
+from cassandra.cluster import NoHostAvailable
+
+from test.pylib.util import wait_for, wait_for_cql_and_get_hosts
+from test.topology.util import wait_for_token_ring_and_group0_consistency, get_coordinator_host
+from test.topology.conftest import skip_mode
+from test.pylib.internal_types import ServerState
+from test.topology_random_failures.cluster_events import CLUSTER_EVENTS, TOPOLOGY_TIMEOUT
+from test.topology_random_failures.error_injections import ERROR_INJECTIONS
+
+if TYPE_CHECKING:
+    from test.pylib.random_tables import RandomTables
+    from test.pylib.internal_types import ServerNum
+    from test.pylib.manager_client import ManagerClient
+    from test.topology_random_failures.cluster_events import ClusterEventType
+
+
+WAIT_FOR_IP_TIMEOUT = 30  # seconds
+
+LOGGER = logging.getLogger(__name__)
+
+
+# Parametrize the test and limit the number of tests to run.
+#
+# Following pytest options can be used to control the number and a sequence of the tests:
+#
+#   `--raft-failure-injections-tests-count' to limit total count of tests from the whole matrix
+#   `--raft-failure-injections-seed' to use some stable order
+#
+# Also, to be able to run same sequence of tests if some new error injection and/or cluster events will be added
+# there are options to exclude these new items:
+#
+#   `--raft-errors-last-index' to limit number of error injections
+#   `--raft-events-last-index' to limit number of cluster events
+#
+def pytest_generate_tests(metafunc: pytest.Metafunc) -> None:
+    tests = list(itertools.product(
+        ERROR_INJECTIONS[:metafunc.config.getoption("--raft-errors-last-index")],
+        CLUSTER_EVENTS[:metafunc.config.getoption("--raft-events-last-index")],
+    ))
+
+    seed = metafunc.config.getoption("--raft-failure-injections-seed")
+    if seed is None:
+        seed = random.randrange(sys.maxsize)
+    LOGGER.info("Effective --raft-failure-injections-seed is %s", seed)
+
+    random.Random(seed).shuffle(tests)
+
+    # Deselect unsupported combinations.  Do it after the shuffle to have the stable order.
+    tests = [
+        (inj, event) for inj, event in tests if inj not in getattr(event, "deselected_random_failures", {})
+    ]
+
+    metafunc.parametrize(
+        ["error_injection", "cluster_event"],
+        tests[:metafunc.config.getoption("--raft-failure-injections-tests-count")],
+    )
+
+
+@pytest.fixture
+async def four_nodes_cluster(manager: ManagerClient) -> list[ServerNum]:
+    LOGGER.info("Booting initial 4-node cluster.")
+    servers = [
+        (await manager.server_add(config={
+            "error_injections_at_startup": [
+                {"name": "raft_server_snapshot_reduce_threshold", "one_shot": True}
```

Comment:
This error injection was removed in 922e03562948c5833a6e37547b2e57b119602ae7 and replaced with more generic one